### PR TITLE
feat: adding questions for General CS Quiz

### PR DESCRIPTION
**Pull request by Markson17**
## Summary of changes

I added unique questions to the list of General CS Quiz, questions on DNS, firewall, OOP, encapsulation and polymorphism

## Checklist

- [x] I have read and followed the [contribution guidelines](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CONTRIBUTING.md).
- [x] I have read through the [Code of Conduct](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CODE_OF_CONDUCT.md) and agree to abide by the rules.
- [x] This PR is for one of the available issues and is not a PR for an issue already assigned to someone else.
- [x] My PR title has a short descriptive name so the maintainers can get an idea of what the PR is about.
- [x] I have provided a summary of my changes.

closes #XXXXX

### CODE REVIEW

File: src/data/general-cs-quiz.ts [hunk 1]
```diff
@@ -1881,7 +1881,52 @@ const generalCSQuiz = [
...
+  },
+  {
+    Question: "What is the primary purpose of the Domain Name System (DNS) in networking?",
+    Answer: "To translate human-readable domain names to IP addresses",
+    Distractor1: "To encrypt data transmitted over the network",
+    Distractor2: "To manage and allocate network bandwidth",
+    Distractor3: "To monitor network traffic for security threats",
+    Explanation: "The Domain Name System (DNS) is responsible for converting domain names, which are easy for humans to remember, into IP addresses that computers use to identify each other on the network.",
+    Link: "https://www.cloudflare.com/learning/dns/what-is-dns/"
+  },
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:1893]: This question would be better in the IT quiz instead of the cs quiz

File: src/data/general-cs-quiz.ts [hunk 2]
```diff
@@ -1881,7 +1881,52 @@ const generalCSQuiz = [
...
-  }
+  },
+  {
+    Question: "What is the primary purpose of the Domain Name System (DNS) in networking?",
+    Answer: "To translate human-readable domain names to IP addresses",
+    Distractor1: "To encrypt data transmitted over the network",
+    Distractor2: "To manage and allocate network bandwidth",
+    Distractor3: "To monitor network traffic for security threats",
+    Explanation: "The Domain Name System (DNS) is responsible for converting domain names, which are easy for humans to remember, into IP addresses that computers use to identify each other on the network.",
+    Link: "https://www.cloudflare.com/learning/dns/what-is-dns/"
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:1892]: We only allow links from freecodecamp sources like the news publication or snippets from freecodecamp youtube videos. we also accept links from official documentation.

we do not accept links from other resources like cloudflare. please use another source.

File: src/data/general-cs-quiz.ts [hunk 3]
```diff
@@ -1881,7 +1881,52 @@ const generalCSQuiz = [
...
+  },
+  {
+    Question: "Which data structure uses LIFO (Last In, First Out) principle?",
+    Answer: "Stack",
+    Distractor1: "Queue",
+    Distractor2: "Linked List",
+    Distractor3: "Hash Table",
+    Explanation: "A stack is a data structure that follows the Last In, First Out (LIFO) principle, meaning the last element added to the stack will be the first one to be removed.",
+    Link: "https://www.geeksforgeeks.org/stack-data-structure/"
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:1901]: We only allow links from freecodecamp sources like the news publication or snippets from freecodecamp youtube videos. we also accept links from official documentation.

we do not accept links from other resources like geeks for geeks

File: src/data/general-cs-quiz.ts [hunk 4]
```diff
@@ -1881,7 +1881,52 @@ const generalCSQuiz = [
...
+    Question: "What is the primary purpose of the Domain Name System (DNS) in networking?",
+    Answer: "To translate human-readable domain names to IP addresses",
+    Distractor1: "To encrypt data transmitted over the network",
+    Distractor2: "To manage and allocate network bandwidth",
+    Distractor3: "To monitor network traffic for security threats",
+    Explanation: "The Domain Name System (DNS) is responsible for converting domain names, which are easy for humans to remember, into IP addresses that computers use to identify each other on the network.",
+    Link: "https://www.cloudflare.com/learning/dns/what-is-dns/"
+  },
+  {
+    Question: "Which data structure uses LIFO (Last In, First Out) principle?",
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:1895]:
```suggestion
    Question: "Which data structure uses the LIFO (Last In, First Out) principle?",
```

File: src/data/general-cs-quiz.ts [hunk 5]
```diff
@@ -1881,7 +1881,52 @@ const generalCSQuiz = [
...
+    Link: "https://www.geeksforgeeks.org/stack-data-structure/"
+  },
+  {
+    Question: "In object-oriented programming, what is encapsulation?",
+    Answer: "The concept of wrapping data and methods that operate on the data within a single unit or class",
+    Distractor1: "The ability of a function to call itself",
+    Distractor2: "The ability to define multiple methods with the same name",
+    Distractor3: "The concept of using symbols to represent operations and data",
+    Explanation: "Encapsulation is an object-oriented programming principle that bundles the data (attributes) and methods (functions) that operate on the data into a single unit or class, and restricts direct access to some of the object's components.",
+    Link: "https://www.geeksforgeeks.org/encapsulation-in-java/"
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:1910]: We only allow links from freecodecamp sources like the news publication or snippets from freecodecamp youtube videos. we also accept links from official documentation.

we do not accept links from other resources like geeks for geeks

File: src/data/general-cs-quiz.ts [hunk 6]
```diff
@@ -1881,7 +1881,52 @@ const generalCSQuiz = [
...
+  },
+  {
+    Question: "What is the purpose of a firewall in computer networks?",
+    Answer: "To monitor and control incoming and outgoing network traffic based on predetermined security rules",
+    Distractor1: "To compress data before transmission",
+    Distractor2: "To increase the speed of data transfer",
+    Distractor3: "To balance the load between multiple servers",
+    Explanation: "A firewall is a network security device that monitors and filters incoming and outgoing network traffic based on an organization's previously established security policies.",
+    Link: "https://www.cisco.com/c/en/us/products/security/firewalls/what-is-a-firewall.html"
+  },
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:1920]: this question should be in the it quiz instead

File: src/data/general-cs-quiz.ts [hunk 7]
```diff
@@ -1881,7 +1881,52 @@ const generalCSQuiz = [
...
+  {
+    Question: "What is polymorphism in the context of object-oriented programming?",
+    Answer: "The ability of different objects to respond uniquely to the same method call",
+    Distractor1: "The process of defining a new class based on an existing class",
+    Distractor2: "The concept of hiding the internal state of an object",
+    Distractor3: "The ability to create multiple methods with the same name but different implementations",
+    Explanation: "Polymorphism is an object-oriented programming principle that allows objects of different classes to be treated as objects of a common super class. It enables a single method to behave differently based on the object that it is acting upon.",
+    Link: "https://www.geeksforgeeks.org/polymorphism-in-java/"
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:1928]: We only allow links from freecodecamp sources like the news publication or snippets from freecodecamp youtube videos. we also accept links from official documentation.

we do not accept links from other resources like geeks for geeks

File: src/data/general-cs-quiz.ts [hunk 8]
```diff
@@ -1881,7 +1881,52 @@ const generalCSQuiz = [
...
+    Link: "https://www.geeksforgeeks.org/encapsulation-in-java/"
+  },
+  {
+    Question: "What is the purpose of a firewall in computer networks?",
+    Answer: "To monitor and control incoming and outgoing network traffic based on predetermined security rules",
+    Distractor1: "To compress data before transmission",
+    Distractor2: "To increase the speed of data transfer",
+    Distractor3: "To balance the load between multiple servers",
+    Explanation: "A firewall is a network security device that monitors and filters incoming and outgoing network traffic based on an organization's previously established security policies.",
+    Link: "https://www.cisco.com/c/en/us/products/security/firewalls/what-is-a-firewall.html"
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:1919]: We only allow links from freecodecamp sources like the news publication or snippets from freecodecamp youtube videos. we also accept links from official documentation.

we do not accept links from other resources like cisco

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please review the requested changes.
Once those changes have been implemented, then we can merge your PR into main. 👍

**Comment by jdwilkin4** [comment]
@Markson17 

Are you still interested in working on this?

**Comment by Markson17** [comment]
Yes I'm still interested in working on it, I would update it later today.

File: src/data/general-cs-quiz.ts (hunk 8 above)
Comment (Markson17) [reply src/data/general-cs-quiz.ts]: Thanks for the feedback, I have replaced the link with an article from the freecodecamp website.

File: src/data/general-cs-quiz.ts (hunk 7 above)
Comment (Markson17) [reply src/data/general-cs-quiz.ts]: Thanks for the feedback, I have replaced the link with an article from the freecodecamp website.

File: src/data/general-cs-quiz.ts (hunk 6 above)
Comment (Markson17) [reply src/data/general-cs-quiz.ts]: I have removed the question from the general CS quiz file and will add it to the IT quiz now. Thank you for your feedback!

File: src/data/general-cs-quiz.ts (hunk 4 above)
Comment (Markson17) [reply src/data/general-cs-quiz.ts]: I have added the suggested change, thanks for the feedback!

File: src/data/general-cs-quiz.ts (hunk 3 above)
Comment (Markson17) [reply src/data/general-cs-quiz.ts]: Thanks for the feedback, I have replaced the link with an article from the freecodecamp website.

File: src/data/general-cs-quiz.ts (hunk 2 above)
Comment (Markson17) [reply src/data/general-cs-quiz.ts]: Thanks for the feedback, I have replaced the link with an article from the freecodecamp website.

File: src/data/general-cs-quiz.ts (hunk 1 above)
Comment (Markson17) [reply src/data/general-cs-quiz.ts]: I have removed the question from the general CS quiz file and will add it to the IT quiz now. Thank you for your feedback!

**Review by jdwilkin4 — APPROVED** [review]

**Pull request merged**